src/anndata/experimental/backed/_lazy_arrays.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+import pandas as pd
+
+from anndata._core.index import _subset
+from anndata._core.views import as_view
+from anndata.compat import H5Array, ZarrArray
+
+from ..._settings import settings
+from ._compat import BackendArray, DataArray, ZarrArrayWrapper
+from ._compat import xarray as xr
+
+if TYPE_CHECKING:
+    from anndata._core.index import Index
+
+
+K = TypeVar("K", H5Array, ZarrArray)
+
+
+class ZarrOrHDF5Wrapper(ZarrArrayWrapper, Generic[K]):
+    def __init__(self, array: K):
+        if isinstance(array, ZarrArray):
+            return super().__init__(array)
+        self._array = array
```

Comment:
you don’t call `super().__init__()` otherwise? Are you sure you’re setting all fields?

Reply:
I added a check against their `__slots__` now